lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -78,6 +79,16 @@ def load
         # re-raise any exception that occurred when reading the metadata
         raise_metadata_error!
 
+        load_all_unignored_files
+
+        # TODO:
+        # 1. Can we safely pick out the segment files with fnmatch against the
+        #    file names in all_unignored_files (?)
+        # 2. If so, can we avoid running chefignore twice by running it against
+        #    all_unignored_files before segmenting?
+        #
+        # See: https://shane.io/2014/07/13/sobbing-with-ruby-file-globbing.html
```

Comment:
I think that's the algorithm I'd prefer to see.  Construct all the files, apply chefignore, then extract the segmented files from the superset...